src/index.d.ts

```diff
@@ -31,46 +51,35 @@ export declare function ThemeProvider<
   }
 >(props: T): JSX.Element;
 export declare function useTheme(): DefaultTheme;
-type Tagged<T> = <P>(
-  args_0:
-    | string
-    | TemplateStringsArray
-    | CSSAttribute
-    | ((
-        props: P &
-          T & {
-            theme?: DefaultTheme;
-            as?: string | number | symbol | undefined;
-            class?: any;
-            children?: any;
-          }
-      ) => string | CSSAttribute),
-  ...args_1: (
-    | string
-    | number
-    | ((
-        props: P &
-          T & {
-            theme?: DefaultTheme;
-            as?: string | number | symbol | undefined;
-            class?: any;
-            children?: any;
-          }
-      ) => string | number | CSSAttribute | undefined)
-  )[]
-) => ((props: P & T) => JSX.Element) & {
-  class: (props: P & T) => string;
+export interface ThemeProp {
+  theme?: DefaultTheme;
+}
```

Comment:
This is would also us to override the props in a project with a theme with something like this:

```
// in a styled.d.ts file
declare module "solid-styled-components" {
  export interface DefaultTheme {
    // ...
  }

  // The theme will always be present
  export interface ThemeProp {
    theme: DefaultTheme;
    //   ^
    // The `?` was removed 
  }
```

We won't need to make sure that the theme exist on the `props` e.g.

```
const Button = styled("button")((props) => ({
  color: props.theme.typography.color,
  //          ^
  // we won't need to check if the theme is present or not
);
```

Reply:
I could extract this in an other PR if needed.

Reply:
This is not possible with the current version of TypeScript. I've added a comment on an existing issue on the TypeScript repo: https://github.com/microsoft/TypeScript/issues/33699#issuecomment-1275535308

Reply:
For now using the non-null assertion operator should do the trick:
```tsx
styled("span")`
  color: ${(props) => props.theme!.palette}
`;
```